core/domain/collection_services.py

```diff
@@ -780,7 +914,12 @@ def _save_collection(committer_id, collection, commit_message, change_list):
             collection_node.to_dict() for collection_node in collection.nodes
         ]
     }
-    collection_model.node_count = len(
+    # Here, collection_model is an instance of `CollectionModel` and
+    # `CollectionModel` does not contain any `node_count` attribute, but
+    # below we are defining a `node_count` attribute which causes MyPy to
+    # throw an error ("CollectionModel" has no attribute "node_count").
+    # Thus to avoid the error, we used an ignore here.
+    collection_model.node_count = len(  # type: ignore[attr-defined]
```

Comment:
This seems like a bug, why do we add `node_count` when it is not defined on the model?

Reply:
After defining `node_count` we are just committing the model and not using `node_count` in further function calls. Also, after removing the node_count line tests are working fine.

Reply:
Yeah, if everything works and it is not used elsewhere then remove it.

Reply:
Removed!